File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.cc
#include <iostream>
using namespace std;

int gcd(int a, int b) {
	for(int i=min(a,b);i>0;i--){
		if (a % i == 0 && b % i == 0) {
			return i;
		}
	}
	return 1;
}

int main() {
	int a, b;
	cin >> a >> b;
	int d = gcd(a, b);
	cout << d << '\n' << a * b / d;
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public static int gcd(int a, int b){
        if(a%b==0) return b;
        System.out.println(a+" "+b);
        return gcd(b, a%b);
    }
    public int[] solution(int n, int m) {
        int a;
        if(n>=m) a = gcd(n, m);
        else a = gcd(m, n);
        return new int[]{a, n*m/a};
    }
}